modules/gapi/src/backends/cpu/gcpubackend.cpp

```diff
@@ -176,8 +169,37 @@ void cv::gimpl::GCPUExecutable::setupKernelStates()
     }
 }
 
+void cv::gimpl::GCPUExecutable::makeReshape() {
+    // Prepare the execution script
+    m_script.clear();
+    for (auto &nh : m_opNodes) {
+        m_script.push_back({nh, GModel::collectOutputMeta(m_gm, nh)});
+    }
+
+    // Preallocate internal mats
+    for (auto& nh : m_dataNodes) {
+        const auto& desc = m_gm.metadata(nh).get<Data>();
+        if (desc.storage == Data::Storage::INTERNAL && desc.shape == GShape::GMAT) {
+            const auto mat_desc = util::get<cv::GMatDesc>(desc.meta);
+            auto& mat = m_res.slot<cv::Mat>()[desc.rc];
+            createMat(mat_desc, mat);
+        }
+    }
+}
+
+void cv::gimpl::GCPUExecutable::reshape(ade::Graph&, const GCompileArgs&) {
```

Comment:
I think, it's needed to handle `GCompileArgs` as well, because `stateful kernels` use them:
https://github.com/opencv/opencv/blob/4.x/modules/gapi/src/backends/cpu/gcpubackend.cpp#L175

I'd propose to add test on that case

Reply:
Done